inference-engine/src/vpu/common/src/ngraph/operations/static_shape_loop.cpp

```diff
@@ -23,22 +23,28 @@ void StaticShapeLoop::validate_and_infer_types() {
 
     Loop::validate_and_infer_types();
 
-    const auto maxIterationsCountEstimation = ngraph::maximum_value(input_value(0).get_node_shared_ptr());
-    NODE_VALIDATION_CHECK(this, maxIterationsCountEstimation.first,
-        "Encountered a loop for which upper-bound estimation for iterations count ", input_value(0), " failed");
-    const auto& maxIterationsCount = maxIterationsCountEstimation.second;
-    NODE_VALIDATION_CHECK(this, maxIterationsCount > 0, "Encountered a loop with non-positive upper-bound estimation for iterations count ",
-        maxIterationsCountEstimation.second);
-
-    const auto& body = get_function();
-    for (const auto& outputDescription : get_output_descriptions()) {
-        if (const auto& concatOutputDescription = ngraph::as_type_ptr<ngraph::op::util::SubGraphOp::ConcatOutputDescription>(outputDescription)) {
-            const auto& bodyOutput = body->output(concatOutputDescription->m_body_value_index);
-            const auto& axis = concatOutputDescription->m_axis;
+    ngraph::PartialShape numIterAsShape;
+    if (!ngraph::evaluate_as_partial_shape(input_value(0), numIterAsShape)) {
+        NODE_VALIDATION_CHECK(this, false,
+                              "Encountered a loop for which upper-bound estimation for iterations count ",
+                              input_value(0), " failed");
+    }
+
+    const auto &maxIterationsCount = numIterAsShape[0].get_max_length();
+    NODE_VALIDATION_CHECK(this, maxIterationsCount > 0,
+                          "Encountered a loop with non-positive upper-bound estimation for iterations count ",
+                          maxIterationsCount);
+
+    const auto &body = get_function();
+    for (const auto &outputDescription : get_output_descriptions()) {
+        if (const auto &concatOutputDescription = ngraph::as_type_ptr<ngraph::op::util::SubGraphOp::ConcatOutputDescription>(
+                outputDescription)) {
```

Comment:
I'd suggest to avoid changes just for the codestyle here

Reply:
Fixed.